### PR TITLE
Deal endpoint changes

### PR DESCRIPTION
**Pull request by jenssegers**

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -849,19 +858,17 @@ Get details for a single deal.
                         + _type: `customField`
                         + id: 123 (string)
                     + value: `092980616`
-        + phase_changes (object)
+        + phase_history/phases/history (object)
```
Comment (stivni) [inline apiary.apib]: phase_history++

File: src/deals.apib [hunk 2]
```diff
@@ -173,6 +182,29 @@ Change the phase of a deal.
 
 + Response 204
 
+### deals.won [POST /deals.won]
```
Comment (stivni) [inline src/deals.apib]: PP : given we added this, I'd go for `move` instead of `changePhase`too

File: src/deals.apib [hunk 3]
```diff
@@ -152,8 +160,9 @@ Update a deal.
         + source: `Google Adwords` (string, optional)
```
Comment (stivni) [inline src/deals.apib]: should be a reference to a dealSource, right?

File: src/deals.apib [hunk 4]
```diff
@@ -45,8 +51,7 @@ Get a list of deals.
                     + id: 1 (string)
                 + created_at: `2017-05-09T11:25:11+00:00` (string)
                 + last_activity_at: `2017-05-09T11:30:58+00:00` (string)
-                + won_at: `2017-05-09T11:31:30+00:00` (string)
-                + lost_at: `2017-05-09T11:31:30+00:00` (string)
+                + closed_at: `2017-05-09T11:31:30+00:00` (string)
                 + reason_lost: `No interest` (string)
```
Comment (stivni) [inline src/deals.apib]: PP : `lost_reason`

File: src/deals.apib [hunk 5]
```diff
@@ -32,11 +32,17 @@ Get a list of deals.
             + (object)
                 + id: `164938` (string)
                 + title: `Interesting deal` (string)
+                + status: `won` (enum[string])
```
Comment (stivni) [inline src/deals.apib]: Q : we miss the sequence number too, right?
I think in projects, we called it the `reference`...

File: apiary.apib [hunk 6]
```diff
@@ -824,22 +829,26 @@ Get details for a single deal.
...
+                + won
+                + lost
         + customer (object)
             + _type: `contact` (string)
             + id: 123 (string)
-        + probability: `0.5` (number)
+        + estimated_value (Money)
+        + estimated_closing_date: `2017-05-09` (string)
+        + estimated_probability: `0.5` (number)
         + phase (object)
```
Comment (stivni) [inline apiary.apib]: PP : `current_phase`

File: apiary.apib [hunk 7]
```diff
@@ -849,19 +858,17 @@ Get details for a single deal.
...
-                    + new_phase (object)
-                        + _type: dealPhase (string)
-                        + id: `2` (string)
-                    + changed_at: `2017-05-01 12:00:00` (string)
-                    + changed_by (object)
+                    + started_by (object)
                         + _type: user (string)
                         + id: `5` (string)
+                    + started_at: `2017-05-01 12:00:00` (string)
+                    + ended_at: `2017-05-01 12:00:00` (string)
```
Comment (stivni) [inline apiary.apib]: NTH : `ended_by`

File: apiary.apib [hunk 8]
```diff
@@ -824,22 +829,26 @@ Get details for a single deal.
...
         + phase (object)
             + _type: dealPhase (string)
             + id: `5` (string)
         + responsible_user (object)
             + _type: user (string)
             + id: 17 (string)
-        + value (Money)
-        + probability: `0.75` (number)
         + created_at: `2017-05-09T11:25:11+00:00` (string)
         + last_activity_at: `2017-05-09T11:30:58+00:00` (string)
```
Comment (stivni) [inline apiary.apib]: PP : not sure about the name... we call it `update_at` in other places

File: apiary.apib [hunk 9]
```diff
@@ -926,6 +935,29 @@ Change the phase of a deal.
...
+
++ Request (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + id: `1` (string, required)
+        + reason: `Bad product fit` (string, optional)
+
++ Response 204
+
 ### dealPhases.list [GET /dealPhases.list]
```
Comment (stivni) [inline apiary.apib]: NTH : think about this endpoint and structure when we would move to multiple deal flows...

File: src/deals.apib [hunk 10]
```diff
@@ -128,8 +135,9 @@ Create a new deal for a customer.
         + department_id: `1` (string, optional)
```
Comment (stivni) [inline src/deals.apib]: Q : create and update have a `department_id`, but the list and info don't...

File: src/deals.apib (hunk 3 above)
Comment (jenssegers) [reply src/deals.apib]: it's kinda the same as with tags, the source is the reference.

Comment (stivni) [reply src/deals.apib]: we should rediscussed this be sure about it... but I'll not let it block the merge

 :ok: by @stivni on 12/10/2017 22:30:23

File: src/deals.apib (hunk 10 above)
Comment (stivni) [reply src/deals.apib]: TIL : the department is saved on the quotation level

Comment (stivni) [reply src/deals.apib]: TIL : we store the deal and quotation in the same record in the db

File: src/deals.apib [hunk 11]
```diff
@@ -40,19 +41,20 @@ Get a list of deals.
...
                     + _type: `dealPhase`
                     + id: 1 (string)
                 + responsible_user (object)
                     + _type: `user`
                     + id: 1 (string)
-                + created_at: `2017-05-09T11:25:11+00:00` (string)
-                + last_activity_at: `2017-05-09T11:30:58+00:00` (string)
                 + closed_at: `2017-05-09T11:31:30+00:00` (string)
-                + reason_lost: `No interest` (string)
+                + lost_reason: `No interested` (string)
```
Comment (stivni) [inline src/deals.apib]: reference to entity?

File: src/deals.apib [hunk 12]
```diff
@@ -61,6 +63,8 @@ Get a list of deals.
                                 + _type: `customField`
                                 + id: 123 (string)
                             + value: `092980616`
+                + created_at: `2017-05-09T11:25:11+00:00` (string)
+                + update_at: `2017-05-09T11:30:58+00:00` (string)
```
Comment (stivni) [inline src/deals.apib]: updated_at

File: apiary.apib (hunk 9 above)
Comment (stivni) [reply apiary.apib]: if we have a separate endpoint for flows, we can always refer to the flow from the phase... so should work...

 :ok: by @stivni on 18/10/2017 19:14:20

**Pull request merged**